### PR TITLE
Refactor menu, table of contents dropdowns

### LINKED ISSUE #409: Refactor top nav to include dropdown menu
The dropdown menu from #406 will be done in this separate ticket instead. There is extensive refactoring required to make this dropdown clean and performant.

## Steps
- Remove CA Web Standard Template
- Test/Research/Decision: How much of the app breaks without it
- Remedy app failures
- Remove git submodule
- Add Bootstrap back
- Rewrite nav and footerwith standard Bootstrap nav

## Options
- Use Bootstrap sass files _in_ the app
- package.json
- Remove tabulator as a git submodule as well (future refactor)

## Testing thus far

App without CA Web State Template SCSS or JS:
image
image
image

Mainly the nav, footer, button colors are broken. The Provider map, Tabulator table and rest of the app work fine.

### PR DESCRIPTION
**Pull request by machikoyasuda**
closes #409 

## What this PR does

- [x] Remove CA State Web Template git submodule
- [x] Remove CA State Web Template SCSS files from index.scss
- [x] Add Bootstrap CSS CDN instead
- [x] Rewrite nav with dropdown.
- [x] Rewrite table of contents mobile dropdown. No more JS.

## Nav

image
image

- Uses Bootstrap dropdown: https://getbootstrap.com/docs/5.2/components/dropdowns/
- The current page is highlighted in `#f5f9fb`. Fixes Liquid logic so that current page detects new How To pages.
- Reduces some logic necessary to create a fixed nav only on Desktop

## Table of contents mobile nav

image
image

- Uses a Bootstrap dropdown https://getbootstrap.com/docs/5.2/components/dropdowns/ and `position: sticky`
- The nav will always open down, even if it doesn't fully fit vertically
- No more fixed nav on Mobile on the How To pages (as the designs originally intended)

## Misc. CSS fixes from the Bootstrap version upgrade

- Uses CSS Variables to declare site-wide variables (colors, link color/styles)
- Table CSS utility classes (`left`/`right` changes to `end`/`start`)

## How to test
- Test nav, toc nav on desktop and mobile
- Test the _whole_ site for any other regressions

### CODE REVIEW

**Comment by machikoyasuda** [comment]
@angela-tran @thekaveman This is ready for code review.

@segacy1 How do the two different navs (How To page, main top nav) look on Desktop and Mobile? The How To page nav now has the correct originally designed function (no more main top nav sticky), and the main top nav now has a Products dropdown.

**Comment by segacy1** [comment]
Hi @machikoyasuda, two small things: 

1. Could "contactless payment" in the products dropdown go to /go-contactless instead of how-to/go-contactless?
2. Can we change "View Contracts" to "Contracts"

Otherwise, this looks great to me! Thank you
cc: @mrose914

**Comment by machikoyasuda** [comment]
Fixed: @segacy1 

1.     "contactless payment" in the products dropdown now goes to `/go-contactless`
2.     `"View Contracts"` changed to `"Contracts"`

image

File: src/_includes/site-header.html
Comment (angela-tran) [inline src/_includes/site-header.html:2]: Did you mean to remove the skip link? I think we should keep it, right?

File: src/_includes/site-header.html
Comment (angela-tran) [inline src/_includes/site-header.html]: I noticed that the dropdown menu has a lot of spacing between it and the nav-item. 

| Screenshot | Dev console |
|---|---|
| image | image |


Maybe we could add a style rule to bring them closer? Something like this?

| Screenshot | Dev console |
|---|---|
| image | image |

**Review by angela-tran — COMMENTED** [review]
I've finished reading through all the commits and left two minor comments. Impressive refactor!

I'll now go through the whole site and visually test for regressions.

File: src/_includes/site-header.html
Comment (segacy1) [reply src/_includes/site-header.html]: Agreed on the spacing for the nav! Otherwise, marking this as approved. Thank you!

**Review by segacy1 — APPROVED** [review]

**Comment by angela-tran** [comment]
Here are some regressions on Desktop I noticed. They are all pretty minor styling things, so I'm not sure if they should block this PR from being merged in or if we want to create separate issues for these so they can be fixed later.

### Focus rings are not showing after clicking (e.g. on buttons, links, nav items)
mobimart-414-button-focus-ring-regression


### Form fields styling 
Bold text and red asterisk are now just normal text and black asterisk.

mobimart-414-form-styling-regression


### Nav bar scroll behavior
When I scroll down, the nav bar doesn't get smaller for me.
mobimart-414-nav-get-smaller-on-scroll-regression

### Nav item hover style
The blue hover style is not showing for me on the nav items.
mobimart-414-nav-item-hover-regression

### Left spacing for nav logo
The logo used to be aligned with the content below but now is slightly to the left of that.
mobimart-414-nav-logo-left-spacing-regression

**Comment by angela-tran** [comment]
@thekaveman Whenever you have time, could you help us with determining priority for fixing the small regressions mentioned above?

File: src/_layouts/default.html
Comment (machikoyasuda) [inline src/_layouts/default.html:36]: Giving this a closer look: I _think_ the animate.css file can also be removed. https://animate.style/ I don't see any CSS classes that start with`animate` in the codebase.

Comment (machikoyasuda) [reply src/_layouts/default.html:36]: CaGov.ttf file can also be removed.

**Comment by machikoyasuda** [comment]
Found another regression:
image

- [ ] Change `right` to `end`
- [ ] Find another SVG (X) icon

**Comment by thekaveman** [comment]
> @thekaveman Whenever you have time, could you help us with determining priority for fixing the small regressions mentioned above?

I think most of the regressions that @angela-tran found don't need to block this going live, we can take them on the backlog and fix as we go forward. Seems like none of them break the UI/UX terribly, mostly nice-to-haves. 

Maybe the logo spacing one is an easier fix that could be added?

File: src/_sass/_variables.scss
Comment (thekaveman) [inline src/_sass/_variables.scss:3]: I wonder if we can get rid of these `!important;` declarations, as noted from the removed comment, they were mostly to ensure overrides of the CA State Template worked as expected.

File: src/_sass/index.scss
Comment (thekaveman) [inline src/_sass/index.scss:6]: I think this is the only other reason we have a submodule. If we can get rid of this import, I think we can get rid of that other submodule as well.

Comment (machikoyasuda) [reply src/_sass/index.scss]: I attempted to remove this submodule and replace it with the CDN links, but it breaks the CSS _so_ much. I would rather not remove it at the time being.

Comment (thekaveman) [reply src/_sass/index.scss:6]: Yikes! Sorry about that. I guess we are relying on it more than I thought...

**Comment by machikoyasuda** [comment]
Ready for re-review @angela-tran 

## New in this version of the PR:
- Skip Link is back.
- Nav on Desktop: Has the hover effects, focus rings. Icon is moved over to the left a bit to align with the rest of the app's container. Nav's dropdown is now closer to the Products link.
- All links now have Focus rings.
- Contracts table regressions are now fixed.
- Form field styling: The Required label is now Red again.
- The X in the Contracts table filter pill is now a Bootstrap icon. CAGov.ttf font file and requirement removed.

The only thing I did not do was the top nav size change on scroll. This can be added to backlog or go back to design to see if we really need it. 

## Pages to test
- Contracts table
- Provider map
- Contact form
- Desktop nav: Focus, hover, dropdown
- Desktop nav: Brand logo aligned
- Skip nav

**Comment by machikoyasuda** [comment]
Note: Found another small regression - this SVG-in-CSS icon is not showing up for some reason. I'd like to make a ticket for this and fix it in a PR after this one is merged. There is upcoming redesigns for Provider Map coming soon too. @angela-tran @thekaveman 
image

**Review by angela-tran — APPROVED** [review]
Verified these changes:
- [x] Focus rings are back
- [x] Form field styling is back
- [x] Nav item hover style is back
- [x] Left spacing for nav logo is good
- [x] Skip link is back

**Comment by machikoyasuda** [comment]
Note: This and subsequent PRs have _not_ been successfully deployed to production. Currently investigating.

Related: https://answers.netlify.com/t/jekyll-build-failing-today-exact-source-built-successfully-yesterday/86038
https://answers.netlify.com/t/build-fails-could-not-find-bundler/86045

**Comment by thekaveman** [comment]
@machikoyasuda @mrose914 I re-ran the deploy from within Netlify, and it succeeded. It was a single deploy from `HEAD`, so it contained all the 3 of the PRs/deploys that failed yesterday.

**Pull request merged**
